### PR TITLE
Update to keycloak 20.0.0

### PR DESCRIPTION
**Pull request by Guy-Incognito**
I still need to update e2e tests

### CODE REVIEW

**Comment by Guy-Incognito** [comment]
@daniel-frak  Rady for your review ;)

File: docker/e2e/cypress/integration/migrating_users.spec.js [hunk 1]
```diff
@@ -48,41 +48,28 @@ describe('user migration plugin', () => {
...
+        cy.get('#kc-forgot-pw-switch')
             .uncheck({force: true})
             .check({force: true});
-
-        cy.get('button').contains('Save').click();
     }
 
     function configureMigrationPlugin() {
-        visitUserFederationPage();
+        // TODO delete existing configured plugins
```
Comment (daniel-frak) [inline docker/e2e/cypress/integration/migrating_users.spec.js:64]: Could you elaborate on this TODO?

File: docker/e2e/cypress/integration/migrating_users.spec.js [hunk 2]
```diff
@@ -104,22 +91,17 @@ describe('user migration plugin', () => {
...
-            cy.get('#lastName').clear().type(ADMIN_USERNAME);
-            cy.get('form[name="userForm"]').get('button[type="submit"]').contains('Save').click({force: true});
-        });
+        cy.visit('/admin/master/console/#/master/users');
+        cy.url().should('include', '/admin/master/console/#/master/users');
+        cy.get('.pf-c-input-group').get('input').clear().type('*');
+        cy.get('.pf-c-input-group').get('.pf-c-button.pf-m-control').click();
+        cy.get('table').get('td[data-label="Username"]').get('a').contains(ADMIN_USERNAME).click();
+        cy.get('#kc-email').clear();
+        cy.wait(500);
```
Comment (daniel-frak) [inline docker/e2e/cypress/integration/migrating_users.spec.js:100]: This seems flakey. Are we sure there is no other way to do this than using `cy.wait(500)`?
If not, then I think a comment could be helpful explaining why the `wait` is needed.

File: docker/e2e/cypress/integration/migrating_users.spec.js [hunk 3]
```diff
@@ -132,6 +114,8 @@ describe('user migration plugin', () => {
         // Wait a while, otherwise Keycloak overrides the inputs randomly
         cy.wait(5000);
 
+        cy.get('#email-address').clear();
+        cy.wait(500);
```
Comment (daniel-frak) [inline docker/e2e/cypress/integration/migrating_users.spec.js:118]: Ditto about `wait`.

File: docker/e2e/cypress/integration/migrating_users.spec.js [hunk 4]
```diff
@@ -170,57 +154,45 @@ describe('user migration plugin', () => {
...
                 cy.get('button').contains('Save').click();
-                cy.wait('@masterPut');
+                cy.get('.pf-c-alert__title').should('contain', "Password policies successfully updated");
             });
     }
 
     function goToPasswordPoliciesPage() {
-        cy.intercept('GET', '/admin/realms/master').as("masterGet");
-        cy.visit('/admin/master/console/#/realms/master/authentication/password-policy');
+        cy.wait(1000);
```
Comment (daniel-frak) [inline docker/e2e/cypress/integration/migrating_users.spec.js:193]: Ditto about `wait`.

**Review by daniel-frak — CHANGES_REQUESTED** [review]

**Comment by daniel-frak** [comment]
Thanks for the contribution!
That's quite an update to the Cypress tests, I'm glad I didn't have to do it :P

I just have some comments regarding the uses of `wait` and a lone `TODO`.

**Comment by daniel-frak** [comment]
LGTM, thanks!

**Pull request merged**